### ISSUE TITLE
xfreerdp regression in 4d885e3a6efd0c45cb5760cb486961b35f21c080

### ISSUE THREAD

**Issue by sss123next** [FreeRDP/FreeRDP#8696]
in commit 4d885e3a6efd0c45cb5760cb486961b35f21c080 introduced problem in xfreerdp client:
```
xfreerdp '/u:shit' /p:QwE123 /v:192.168.1.110 /log-level:DEBUG /sec:tls /cert:ignore /size:1920x1080
[00:58:08:770] [898952:898952] [DEBUG][com.freerdp.client.common] - This is 3.0.0-dev Build configuration: BUILD_TESTING=OFF HAVE_AIO_H=1 HAVE_EXECINFO_BACKTRACE=1 HAVE_EXECINFO_BACKTRACE_SYMBOLS=1 HAVE_EXECINFO_BACKTRACE_SYMBOLS_FD=1 HAVE_EXECINFO_HEADER=1 HAVE_FCNTL_H=1 HAVE_GETLOGIN_R=1 HAVE_GETPWUID_R=1 HAVE_INTTYPES_H=1 HAVE_JOURNALD_H=TRUE HAVE_POLL_H=1 HAVE_PTHREAD_MUTEX_TIMEDLOCK_LIB=1 HAVE_PTHREAD_MUTEX_TIMEDLOCK_LIBS= HAVE_PTHREAD_MUTEX_TIMEDLOCK_SYMBOL=1 HAVE_STDBOOL_H=1 HAVE_STDINT_H=1 HAVE_STRNDUP=1 HAVE_SYSLOG_H=1 HAVE_SYS_EVENTFD_H=1 HAVE_SYS_FILIO_H= HAVE_SYS_SELECT_H=1 HAVE_SYS_SOCKIO_H= HAVE_SYS_TIMERFD_H=1 HAVE_TM_GMTOFF=1 HAVE_UNISTD_H=1 HAVE_UNWIND_H=1 WITH_ADD_PLUGIN_TO_RPATH=OFF WITH_ALSA=ON WITH_CAIRO=ON WITH_CCACHE=ON WITH_CHANNELS=ON WITH_CLANG_FORMAT=ON WITH_CLIENT=ON WITH_CLIENT_AVAILABLE=1 WITH_CLIENT_CHANNELS=ON WITH_CLIENT_CHANNELS_AVAILABLE=1 WITH_CLIENT_COMMON=ON WITH_CLIENT_INTERFACE=OFF WITH_CUPS=OFF WITH_DEBUG_ALL=OFF WITH_DEBUG_CAPABILITIES=OFF WITH_DEBUG_CERTIFICATE=OFF WITH_DEBUG_CHANNELS=OFF WITH_DEBUG_CLIPRDR=OFF WITH_DEBUG_DVC=OFF WITH_DEBUG_EVENTS=OFF WITH_DEBUG_KBD=OFF WITH_DEBUG_LICENSE=OFF WITH_DEBUG_MUTEX=OFF WITH_DEBUG_NEGO=OFF WITH_DEBUG_NLA=OFF WITH_DEBUG_NTLM=OFF WITH_DEBUG_RAIL=OFF WITH_DEBUG_RDP=OFF WITH_DEBUG_RDPDR=OFF WITH_DEBUG_RDPEI=OFF WITH_DEBUG_RDPGFX=OFF WITH_DEBUG_REDIR=OFF WITH_DEBUG_RFX=OFF WITH_DEBUG_RINGBUFFER=OFF WITH_DEBUG_SCARD=OFF WITH_DEBUG_SND=OFF WITH_DEBUG_SVC=OFF WITH_DEBUG_SYMBOLS=OFF WITH_DEBUG_THREADS=OFF WITH_DEBUG_TIMEZONE=OFF WITH_DEBUG_TRANSPORT=OFF WITH_DEBUG_TSG=OFF WITH_DEBUG_TSMF=OFF WITH_DEBUG_TSMF_AVAILABLE=0 WITH_DEBUG_URBDRC=OFF WITH_DEBUG_WND=OFF WITH_DEBUG_X11=OFF WITH_DEBUG_X11_CLIPRDR=OFF WITH_DEBUG_X11_LOCAL_MOVESIZE=OFF WITH_DEBUG_XV=OFF WITH_DSP_EXPERIMENTAL=OFF WITH_DSP_FFMPEG=OFF WITH_DSP_FFMPEG_AVAILABLE=0 WITH_EVENTFD_READ_WRITE=1 WITH_FAAC=OFF WITH_FAAD2=ON WITH_FFMPEG=OFF WITH_FREERDP_DEPRECATED=OFF WITH_FREERDP_DEPRECATED_COMMANDLINE=OFF WITH_FUSE=OFF WITH_GFX_H264=OFF WITH_GPROF=OFF WITH_GSM=OFF WITH_GSSAPI=OFF WITH_INTERNAL_MD4=OFF WITH_INTERNAL_MD5=OFF WITH_IPP=OFF WITH_JPEG=ON WITH_LAME=OFF WITH_LIBRARY_VERSIONING=ON WITH_LIBSYSTEMD=ON WITH_MACAUDIO=OFF WITH_MACAUDIO_AVAILABLE=0 WITH_MANPAGES=ON WITH_MBEDTLS=OFF WITH_NATIVE_SSPI=OFF WITH_OPENCL=OFF WITH_OPENH264=OFF WITH_OPENSLES=OFF WITH_OPENSSL=ON WITH_OSS=OFF WITH_PAM=ON WITH_PCSC=OFF WITH_PKCS11=OFF WITH_POLL=ON WITH_PROFILER=OFF WITH_PROXY=ON WITH_PROXY_APP=ON WITH_PROXY_EMULATE_SMARTCARD=OFF WITH_PROXY_MODULES=ON WITH_PULSE=OFF WITH_RDTK=ON WITH_SAMPLE=OFF WITH_SANITIZE_ADDRESS=OFF WITH_SANITIZE_ADDRESS_AVAILABLE=1 WITH_SANITIZE_MEMORY=OFF WITH_SANITIZE_MEMORY_AVAILABLE=1 WITH_SANITIZE_THREAD=OFF WITH_SANITIZE_THREAD_AVAILABLE=1 WITH_SERVER=ON WITH_SERVER_CHANNELS=ON WITH_SERVER_INTERFACE=ON WITH_SHADOW=ON WITH_SMARTCARD_EMULATE=OFF WITH_SMARTCARD_INSPECT=OFF WITH_SMARTCARD_PCSC=ON WITH_SOXR=OFF WITH_SSE2=ON WITH_SWSCALE=OFF WITH_THIRD_PARTY=OFF WITH_VAAPI=OFF WITH_VAAPI=OFF WITH_VAAPI_AVAILABLE=0 WITH_VALGRIND_MEMCHECK=OFF WITH_VALGRIND_MEMCHECK_AVAILABLE=1 WITH_VERBOSE_WINPR_ASSERT=ON WITH_VIDEO_FFMPEG=OFF WITH_VIDEO_FFMPEG_AVAILABLE=0 WITH_WAYLAND=OFF WITH_WINPR_DEPRECATED=OFF WITH_WINPR_TOOLS=ON WITH_X11=ON WITH_XCURSOR=ON WITH_XDAMAGE=ON WITH_XEXT=ON WITH_XFIXES=ON WITH_XI=ON WITH_XINERAMA=ON WITH_XKBFILE=ON WITH_XRANDR=ON WITH_XRENDER=ON WITH_XSHM=ON WITH_XTEST=ON WITH_XV=ON WITH_ZLIB=ON
Build type:          Release
CFLAGS:               -fPIC -Wall -fvisibility=hidden -Wimplicit-function-declaration -Wredundant-decls -fno-omit-frame-pointer
Compiler:            GNU, 11.3.0
Target architecture: x64

[00:58:08:770] [898952:898953] [DEBUG][com.freerdp.core] - freerdp_connect_begin:freerdp_set_last_error_ex resetting error state
[00:58:08:772] [898952:898953] [DEBUG][com.freerdp.client.x11] - Property 450 does not exist
[00:58:09:507] [898952:898953] [DEBUG][com.freerdp.client.common.cmdline] - loading channelEx rdpdr
[00:58:09:507] [898952:898953] [DEBUG][com.freerdp.client.common.cmdline] - loading channelEx rdpsnd
[00:58:09:507] [898952:898953] [DEBUG][com.freerdp.channels.cliprdr.client] - VirtualChannelEntryEx
[00:58:09:507] [898952:898953] [DEBUG][com.freerdp.client.common.cmdline] - loading channelEx cliprdr
[00:58:09:507] [898952:898953] [DEBUG][com.freerdp.channels.drdynvc.client] - VirtualChannelEntryEx
[00:58:09:507] [898952:898953] [DEBUG][com.freerdp.client.common.cmdline] - loading channelEx drdynvc
[00:58:09:509] [898952:898953] [DEBUG][com.freerdp.primitives] - primitives benchmark result:
[00:58:09:665] [898952:898953] [DEBUG][com.freerdp.primitives] -  * generic= 42
[00:58:09:816] [898952:898953] [DEBUG][com.freerdp.primitives] -  * optimized= 145
[00:58:09:816] [898952:898953] [DEBUG][com.freerdp.primitives] - primitives autodetect, using optimized
[00:58:09:834] [898952:898953] [DEBUG][com.freerdp.core.nego] - Enabling security layer negotiation: TRUE
[00:58:09:834] [898952:898953] [DEBUG][com.freerdp.core.nego] - Enabling restricted admin mode: FALSE
[00:58:09:834] [898952:898953] [DEBUG][com.freerdp.core.nego] - Enabling RDP security: FALSE
[00:58:09:834] [898952:898953] [DEBUG][com.freerdp.core.nego] - Enabling TLS security: TRUE
[00:58:09:834] [898952:898953] [DEBUG][com.freerdp.core.nego] - Enabling NLA security: FALSE
[00:58:09:834] [898952:898953] [DEBUG][com.freerdp.core.nego] - Enabling NLA extended security: FALSE
[00:58:09:834] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_INITIAL --> CONNECTION_STATE_NEGO
[00:58:09:834] [898952:898953] [DEBUG][com.freerdp.core.nego] - state: NEGO_STATE_TLS
[00:58:09:834] [898952:898953] [DEBUG][com.freerdp.core.nego] - Attempting TLS security
[00:58:09:834] [898952:898953] [INFO][com.freerdp.core.proxy] - Parsed proxy configuration: http://192.168.0.254:3128
[00:58:09:834] [898952:898953] [DEBUG][com.freerdp.core.proxy] - 192.168.1.110 => localhost (9)
[00:58:09:834] [898952:898953] [DEBUG][com.freerdp.core.proxy] - 192.168.1.110 => 127.0.0.1 (9)
[00:58:09:834] [898952:898953] [DEBUG][com.freerdp.core.proxy] - 192.168.1.110 => 192.168.0.0/24 (14)
[00:58:09:834] [898952:898953] [DEBUG][com.freerdp.core.proxy] - 192.168.1.110 => sss.chaoslab.ru (15)
[00:58:09:834] [898952:898953] [DEBUG][com.freerdp.core.proxy] - 192.168.1.110 => dark-alexandr.net (17)
[00:58:09:834] [898952:898953] [DEBUG][com.freerdp.core.proxy] - 192.168.1.110 => 192.168.1.0/24 (14)
[00:58:09:834] [898952:898953] [INFO][com.freerdp.core.proxy] - deactivating proxy: 192.168.1.110 [no_proxy=localhost,127.0.0.1,192.168.0.0/24,sss.chaoslab.ru,dark-alexandr.net,192.168.1.0/24,192.168.88.0/24,192.168.14.0/24]
[00:58:09:834] [898952:898953] [DEBUG][com.freerdp.core] - freerdp_tcp_is_hostname_resolvable:freerdp_set_last_error_ex resetting error state
[00:58:09:834] [898952:898953] [DEBUG][com.freerdp.core] - freerdp_tcp_default_connect:freerdp_set_last_error_ex resetting error state
[00:58:09:834] [898952:898953] [DEBUG][com.freerdp.core] - connecting to peer 192.168.1.110
[00:58:09:836] [898952:898953] [DEBUG][com.freerdp.core.nego] - RequestedProtocols: 1
[00:58:09:855] [898952:898953] [DEBUG][com.freerdp.core.nego] - RDP_NEG_RSP::flags = { [0x1f] |EXTENDED_CLIENT_DATA_SUPPORTED|DYNVC_GFX_PROTOCOL_SUPPORTED|RDP_NEGRSP_RESERVED|RESTRICTED_ADMIN_MODE_SUPPORTED|REDIRECTED_AUTHENTICATION_MODE_SUPPORTED }
[00:58:09:855] [898952:898953] [DEBUG][com.freerdp.core.nego] - selected_protocol: 1
[00:58:09:855] [898952:898953] [DEBUG][com.freerdp.core.nego] - Negotiated TLS security
[00:58:09:855] [898952:898953] [DEBUG][com.freerdp.core.nego] - nego_security_connect with PROTOCOL_SSL
[00:58:09:865] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_NEGO --> CONNECTION_STATE_MCS_CREATE_REQUEST
[00:58:09:865] [898952:898953] [DEBUG][com.freerdp.core.gcc] - Sending highColorDepth=0x0018, supportedColorDepths=0x000f, earlyCapabilityFlags=RNS_UD_CS_WANT_32BPP_SESSION|RNS_UD_CS_VALID_CONNECTION_TYPE|RNS_UD_CS_SUPPORT_NETCHAR_AUTODETECT|RNS_UD_CS_SUPPORT_DYNVC_GFX_PROTOCOL|RNS_UD_CS_SUPPORT_HEARTBEAT_PDU| [0x05a2]
[00:58:09:865] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_MCS_CREATE_REQUEST --> CONNECTION_STATE_MCS_CREATE_RESPONSE
[00:58:09:868] [898952:898953] [DEBUG][com.freerdp.core.gcc] - Received EarlyCapabilityFlags=RNS_UD_CS_SUPPORT_STATUSINFO_PDU| [0x0004]
[00:58:09:868] [898952:898953] [DEBUG][com.freerdp.core.gcc] - Server rdp encryption method: NONE
[00:58:09:868] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_MCS_CREATE_RESPONSE --> CONNECTION_STATE_MCS_ERECT_DOMAIN
[00:58:09:868] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_MCS_ERECT_DOMAIN --> CONNECTION_STATE_MCS_ATTACH_USER
[00:58:09:868] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_MCS_ATTACH_USER --> CONNECTION_STATE_MCS_ATTACH_USER_CONFIRM
[00:58:09:870] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_MCS_ATTACH_USER_CONFIRM --> CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST
[00:58:09:870] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST --> CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE
[00:58:09:871] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE --> CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST
[00:58:09:871] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST --> CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE
[00:58:09:872] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE --> CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST
[00:58:09:872] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST --> CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE
[00:58:09:874] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE --> CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST
[00:58:09:874] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST --> CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE
[00:58:09:875] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE --> CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST
[00:58:09:875] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST --> CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE
[00:58:09:876] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE --> CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST
[00:58:09:876] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST --> CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE
[00:58:09:877] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE --> CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST
[00:58:09:877] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_MCS_CHANNEL_JOIN_REQUEST --> CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE
[00:58:09:878] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_MCS_CHANNEL_JOIN_RESPONSE --> CONNECTION_STATE_SECURE_SETTINGS_EXCHANGE
[00:58:09:878] [898952:898953] [DEBUG][com.freerdp.core.info] - Client Info Packet Flags = INFO_MOUSE|INFO_DISABLECTRLALTDEL|INFO_AUTOLOGON|INFO_UNICODE|INFO_MAXIMIZESHELL|INFO_LOGONNOTIFY|INFO_COMPRESSION|INFO_ENABLEWINDOWSKEY|INFO_FORCE_ENCRYPTED_CS_PDU|INFO_LOGONERRORS|INFO_MOUSE_HAS_WHEEL|INFO_NOAUDIOPLAYBACK
[00:58:09:878] [898952:898953] [INFO][com.winpr.timezone] - tzid: Europe/Moscow
[00:58:09:879] [898952:898953] [DEBUG][com.winpr.timezone] - tz: Bias=-180 sn='Russia TZ 2 Standard Time' dln='Russia TZ 2 Daylight Time'
[00:58:09:879] [898952:898953] [ERROR][com.winpr.timezone] - Unable to get current timezone rule
[00:58:09:879] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_SECURE_SETTINGS_EXCHANGE --> CONNECTION_STATE_CONNECT_TIME_AUTO_DETECT_REQUEST
[00:58:09:889] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_CONNECT_TIME_AUTO_DETECT_REQUEST --> CONNECTION_STATE_LICENSING
[00:58:09:889] [898952:898953] [WARN][com.freerdp.core.license] - license binary blob::type BB_ERROR_BLOB, length=0, skipping.
[00:58:09:889] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_LICENSING --> CONNECTION_STATE_CAPABILITIES_EXCHANGE_DEMAND_ACTIVE
[00:58:09:020] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_CAPABILITIES_EXCHANGE_DEMAND_ACTIVE --> CONNECTION_STATE_CAPABILITIES_EXCHANGE_CONFIRM_ACTIVE
[00:58:09:020] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_CAPABILITIES_EXCHANGE_CONFIRM_ACTIVE --> CONNECTION_STATE_FINALIZATION_SYNC
[00:58:09:020] [898952:898953] [DEBUG][com.freerdp.core.rdp] - [CONNECTION_STATE_FINALIZATION_SYNC] reset finalize_sc_pdus
[00:58:09:020] [898952:898953] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x1f size=37 channelId=1009)
[00:58:09:020] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_FINALIZATION_SYNC --> CONNECTION_STATE_FINALIZATION_COOPERATE
[00:58:09:020] [898952:898953] [DEBUG][com.freerdp.core.rdp] - [CONNECTION_STATE_FINALIZATION_COOPERATE] reset finalize_sc_pdus
[00:58:09:020] [898952:898953] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x14 size=41 channelId=1009)
[00:58:09:020] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_FINALIZATION_COOPERATE --> CONNECTION_STATE_FINALIZATION_REQUEST_CONTROL
[00:58:09:020] [898952:898953] [DEBUG][com.freerdp.core.rdp] - [CONNECTION_STATE_FINALIZATION_REQUEST_CONTROL] reset finalize_sc_pdus
[00:58:09:020] [898952:898953] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x14 size=41 channelId=1009)
[00:58:09:020] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_FINALIZATION_REQUEST_CONTROL --> CONNECTION_STATE_FINALIZATION_PERSISTENT_KEY_LIST
[00:58:09:020] [898952:898953] [DEBUG][com.freerdp.core.rdp] - [CONNECTION_STATE_FINALIZATION_PERSISTENT_KEY_LIST] reset finalize_sc_pdus
[00:58:09:020] [898952:898953] [DEBUG][com.freerdp.core.activation] - Persistent Key List: TotalKeyCount: 0 MaxKeyFrag: 2042
[00:58:09:020] [898952:898953] [DEBUG][com.freerdp.core.activation] - persistentKeyList count: 0
[00:58:09:020] [898952:898953] [DEBUG][com.freerdp.core.activation] - numEntriesCache: [0]: 0 [1]: 0 [2]: 0 [3]: 0 [4]: 0
[00:58:09:020] [898952:898953] [DEBUG][com.freerdp.core.activation] - totalEntriesCache: [0]: 0 [1]: 0 [2]: 0 [3]: 0 [4]: 0
[00:58:09:020] [898952:898953] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x2b size=57 channelId=1009)
[00:58:09:020] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_FINALIZATION_PERSISTENT_KEY_LIST --> CONNECTION_STATE_FINALIZATION_FONT_LIST
[00:58:09:020] [898952:898953] [DEBUG][com.freerdp.core.rdp] - [CONNECTION_STATE_FINALIZATION_FONT_LIST] reset finalize_sc_pdus
[00:58:09:020] [898952:898953] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x27 size=41 channelId=1009)
[00:58:09:020] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_FINALIZATION_FONT_LIST --> CONNECTION_STATE_FINALIZATION_CLIENT_SYNC
[00:58:09:021] [898952:898953] [DEBUG][com.freerdp.core.rdp] - recv Synchronize Data PDU (0x1F), length: 22
[00:58:09:021] [898952:898953] [DEBUG][com.freerdp.core.rdp] - [CONNECTION_STATE_FINALIZATION_CLIENT_SYNC] received flag FINALIZE_SC_SYNCHRONIZE_PDU [0x0001]
[00:58:09:021] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_FINALIZATION_CLIENT_SYNC --> CONNECTION_STATE_FINALIZATION_CLIENT_COOPERATE
[00:58:09:021] [898952:898953] [DEBUG][com.freerdp.core.rdp] - recv Control Data PDU (0x14), length: 26
[00:58:09:021] [898952:898953] [DEBUG][com.freerdp.core.rdp] - [CONNECTION_STATE_FINALIZATION_CLIENT_COOPERATE] received flag FINALIZE_SC_CONTROL_COOPERATE_PDU [0x0002]
[00:58:09:021] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_FINALIZATION_CLIENT_COOPERATE --> CONNECTION_STATE_FINALIZATION_CLIENT_GRANTED_CONTROL
[00:58:09:021] [898952:898953] [DEBUG][com.freerdp.core.rdp] - recv Control Data PDU (0x14), length: 26
[00:58:09:021] [898952:898953] [DEBUG][com.freerdp.core.rdp] - [CONNECTION_STATE_FINALIZATION_CLIENT_GRANTED_CONTROL] received flag FINALIZE_SC_CONTROL_GRANTED_PDU [0x0004]
[00:58:09:021] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_FINALIZATION_CLIENT_GRANTED_CONTROL --> CONNECTION_STATE_FINALIZATION_CLIENT_FONT_MAP
[00:58:09:021] [898952:898953] [DEBUG][com.freerdp.core.rdp] - recv Font Map Data PDU (0x28), length: 26
[00:58:09:021] [898952:898953] [DEBUG][com.freerdp.core.rdp] - [CONNECTION_STATE_FINALIZATION_CLIENT_FONT_MAP] received flag FINALIZE_SC_FONT_MAP_PDU [0x0008]
[00:58:09:021] [898952:898953] [DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_FINALIZATION_CLIENT_FONT_MAP --> CONNECTION_STATE_ACTIVE
[00:58:09:024] [898952:898953] [DEBUG][com.freerdp.client.x11] - Virtual core XTEST pointer button device (id: 4, mode: 10)
[00:58:09:024] [898952:898953] [DEBUG][com.freerdp.client.x11] - sss YMDK Melody96 keyboard (sss mod) Mouse button device (id: 9, mode: 12)
[00:58:09:024] [898952:898953] [DEBUG][com.freerdp.client.x11] - sss YMDK Melody96 keyboard (sss mod) Consumer Control button device (id: 11, mode: 7)
[00:58:09:024] [898952:898953] [DEBUG][com.freerdp.client.x11] - RAPOO Rapoo 2.4G Wireless Device button device (id: 13, mode: 9)
[00:58:09:024] [898952:898953] [DEBUG][com.freerdp.client.x11] - RAPOO Rapoo 2.4G Wireless Device Consumer Control button device (id: 15, mode: 7)
[00:58:09:024] [898952:898953] [DEBUG][com.freerdp.client.x11] - Razer Razer Pro Click button device (id: 17, mode: 9)
[00:58:09:024] [898952:898953] [DEBUG][com.freerdp.client.x11] - Razer Razer Pro Click Keyboard button device (id: 18, mode: 7)
[00:58:09:024] [898952:898953] [DEBUG][com.freerdp.client.x11] - ELECOM TrackBall Mouse DEFT Pro TrackBall Mouse button device (id: 21, mode: 12)
[00:58:09:024] [898952:898953] [DEBUG][com.freerdp.client.x11] - ELECOM TrackBall Mouse DEFT Pro TrackBall Consumer Control button device (id: 23, mode: 7)
[00:58:09:024] [898952:898953] [DEBUG][com.freerdp.client.x11] - Kensington Expert Mouse button device (id: 25, mode: 9)
X Error of failed request:  BadValue (integer parameter out of range for operation)
  Major opcode of failed request:  131 (XInputExtension)
  Minor opcode of failed request:  46 ()
  Value in failed request:  0xb
  Serial number of failed request:  68
  Current serial number in output stream:  68

```

as changes in commit related to color deph, it may be somehow related to 10bit display mode

**Comment by akallabeth** [comment]
@sss123next would be nice to know where this is running on.
need to reproduce this to figure out what exactly is breaking.

**Comment by akallabeth** [comment]
@sss123next also, since the error mentiones `XInputExtension`, does it work if you connect fewer input devices?

**Comment by sss123next** [comment]
it's on nixos, freerdp built with following script

you need to set required commit in `rev`
```
let 
  nixpkgsmaster = import <nixpkgsmaster>{};
  nixpkgs2111 = import <nixpkgs2111>{};
  nixpkgs2105 = import <nixpkgs2105>{};
  nixpkgs2009 = import <nixpkgs2009>{};
  nixpkgs2003 = import <nixpkgs2003>{};
  nixpkgs1809 = import <nixpkgs1809>{};
  cmFlag = flag: if flag then "ON" else "OFF";
in
  self: super: {
    freerdp_my = with self.lib;  with super; with xorg; stdenv.mkDerivation rec {
  pname = "freerdp";
#  version = "2023-02-14";
  version = "2022-11-16";

  src = fetchFromGitHub {
    owner = "FreeRDP";
    repo = "FreeRDP";
#    rev = "5f9db5a89c0bb134909afca84bc359df1c58de59";
#    sha256 = "sha256-Jb9jgjCqafT50j6mK2omf96w9I3L3p7FuphXUjhivS0=";
#    rev = "fb3f85779b28ff2d16218dbb393d5c191322a747";
#    sha256 = "sha256-QoBQCpFCpdtHKyJRaL8MfewzTzZmMaVvkxiiYUPJ6gA=";

    rev = "7b6023b3400658aa7b303997a221814daccd44fc";
    sha256 = "sha256-eeC2DTNCYAVbEsawrdco6agmPy+vPbKgNs7kqHyzkno=";

  };

  doCheck = false;

  postPatch = ''
    export HOME=$TMP

    # skip NIB file generation on darwin
    sed -z 's/NIB file generation.*//' -i client/Mac{,/cli}/CMakeLists.txt

    substituteInPlace "libfreerdp/freerdp.pc.in" \
      --replace "Requires:" "Requires: @WINPR_PKG_CONFIG_FILENAME@"
  '' + lib.optionalString (pcsclite != null) ''
    substituteInPlace "winpr/libwinpr/smartcard/smartcard_pcsc.c" \
      --replace "libpcsclite.so" "${lib.getLib pcsclite}/lib/libpcsclite.so"
  '';

  buildInputs = [
    cairo
    faad2
    ffmpeg
    glib
    icu
    libX11
    libXcursor
    libXdamage
    libXdmcp
    libXext
    libXi
    libXinerama
    libXrandr
    libXrender
    libXtst
    libXv
    libjpeg_turbo
    libunwind
    libusb1
    libxkbcommon
    libxkbfile
    openssl
    orc
    pcre2
    zlib
    alsa-lib
    systemd
  ];

  nativeBuildInputs = [ cmake libxslt docbook-xsl-nons pkg-config ];

  # https://github.com/FreeRDP/FreeRDP/issues/8526#issuecomment-1357134746
  cmakeFlags = [
    "-Wno-dev"
    "-DCMAKE_INSTALL_LIBDIR=lib"
    "-DDOCBOOKXSL_DIR=${docbook-xsl-nons}/xml/xsl/docbook"
  ]
  ++ lib.mapAttrsToList (k: v: "-D${k}=${cmFlag v}") {
    BUILD_TESTING = false; # false is recommended by upstream
    WITH_CAIRO = true;
    WITH_CUPS = false;
    WITH_FAAC = false;
    WITH_FAAD2 = true;
    WITH_JPEG = true;
    WITH_OPENH264 = false;
    WITH_OSS = false;
    WITH_PCSC = false;
    WITH_PULSE = false;
    WITH_SERVER = true;
    WITH_VAAPI = false; # false is recommended by upstream
    WITH_X11 = true;
  };

  NIX_CFLAGS_COMPILE = lib.optionals stdenv.isDarwin [
    "-DTARGET_OS_IPHONE=0"
    "-DTARGET_OS_WATCH=0"
    "-include AudioToolbox/AudioToolbox.h"
  ];

  NIX_LDFLAGS = lib.optionals stdenv.isDarwin [
    "-framework AudioToolbox"
  ];

  meta = with lib; {
    description = "A Remote Desktop Protocol Client";
    longDescription = ''
      FreeRDP is a client-side implementation of the Remote Desktop Protocol (RDP)
      following the Microsoft Open Specifications.
    '';
    homepage = "https://www.freerdp.com/";
    license = licenses.asl20;
    maintainers = with maintainers; [ peterhoeg lheckemann ];
    platforms = platforms.unix;
  };
};
```


```
sss: does it work on your end if you change back the line `+ xfc-&gt;depth = 32; // DefaultDepthOfScreen(xfc-&gt;screen);`? that seems to be the relevant change from that commit
```

unfortunately i have not time to test it yet.

**Comment by akallabeth** [comment]
Ok, implemented better `X11` error logging in #8737 
that might give us some more details if the issue occurs again.
